src/Persistence/Search/SearchIndexFieldsBuilder.php

```diff
@@ -25,20 +25,25 @@ public function __construct(
 	 * @return array<string, SearchIndexField>
 	 */
 	public function createFields(): array {
-		$fields = [];
-
-		foreach ( $this->config->getFacets()->asArray() as $facetConfig ) {
-			$fieldType = $this->getFieldTypeForPropertyId( $facetConfig->propertyId );
+		try {
```

Comment:
`createFieldObjects` might be better. Generally words like `Data`, `Object`, and `Info`, are to be avoided, but in this case you make it clear that you are not creating the field in elastic.

Reply:
The return type doc could say something about the key being the name for the index.

Could also create a value object and return `ThatNewType[]`, but not a huge deal